security/selinux_policy_binary.ksy

```diff
@@ -0,0 +1,1615 @@
+meta:
+  id: selinux_policy_binary
+  title: SELinux file policy binary
+  file-extension: bin
+  license: CC0-1.0
+  encoding: ASCII
+  endian: le
+doc-ref:
```

Comment:
It would greatly benefit casual user to know more if we could include at least a brief overview of what this file is all about:

* 1-2 sentences about what SELinux is with links to where to read more.
* which systems/applications are reading it and for which purposes,
* which systems/applications are writing/generating it (I guess it's result of compilation of text-based policies in binary db?)
* if these files are located in the OS in some well-known locations, note where they can be found
* how many different versions of this format exist
* what's the source of truth (e.g. it seems to be transcribed from code, which is the only source of truth?)

This will also make it more SEO-friendly, so people can start finding this when they search for something like "selinux binary format specification".

Reply:
@GreyCat Note: we should probably finally resolve the TODO in https://doc.kaitai.io/ksy_style_guide.html#doc:

> ### 4.1. `doc`

> **TODO**: documentation contents, what should and should no be included

... so that we can just link there when needed, instead of reproducing it on individual PRs.

Reply:
There's no official specs. I'm linking to the official source of thruth, the library code.
I tried to put in a short summary but this whole section could expand into... books ;)
In any case, this is not a specification, just a binary format grammar:
without the whole 'how the data is handled', it's just a thorough description of the file contents.
An actual specification would get in all the different versions and how to handle them individually, and that would be a lot of more than a page of text.